packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -2007,6 +1900,22 @@ function commitWork(current: Fiber | null, finishedWork: Fiber): void {
             commitHydratedContainer(root.containerInfo);
           }
         }
+      } else {
+        // Persistent mode
```

Comment:
This else is for `supportsHydration` so I think maybe you mean `else if (!supportsMutation && supportsPersistence)` or similar.

Reply:
I think this shouldn't be an `else` though because `commitHydratedContainer` call above is shared between mutation _and_ persistence. So I think this branch should be extra code _after_ that.

```js
if (supportsPersistence) {
 ...
}
```

Reply:
Yeah I think when I was tidying up I pattern matched wrong on this and the other change.

Reply:
Fixed: https://github.com/facebook/react/compare/6c608931b7c787fa5d6a6b66f8c7c83608e5a6e5..04d140c1138e3aa9dfdbf02a12faa7b49c28514e